### PR TITLE
SmokeTestWatcherWithSecurityIT: Retry if failures searching .watcher-history

### PR DESCRIPTION
**Pull request by jakelandis**
Catch exception and fail (which allows the assertBusy to retry)
if an error happens from the search against .watcher-history.

Related: #42409
Related: #39169

### CODE REVIEW

**Comment by jakelandis** [comment]
I should note this is the same change that was applied to the non-security variant: https://github.com/elastic/elasticsearch/blob/master/x-pack/qa/smoke-test-watcher/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherTestSuiteIT.java#L218

File: x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java
Comment (dakrone) [inline x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java]: Any particular reason not to use

```suggestion
                logger.info("Failed to perform search of watcher history", e);
```

Here instead so that you get the whole stacktrace?

**Review by dakrone — APPROVED** [review]
LGTM, left one comment (optional if you want to do it or not)

File: x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java
Comment (jakelandis) [reply x-pack/qa/smoke-test-watcher-with-security/src/test/java/org/elasticsearch/smoketest/SmokeTestWatcherWithSecurityIT.java]: thanks @dakrone , I updated this and the other code (from which this came) to preserve the stack trace. https://github.com/elastic/elasticsearch/pull/43781/commits/e448050afabf469ae422ebdd11920a163f0c0573

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
holding off on backporting since this test is currently muted in the older pipelines.  Trying to stabilize this test in master before backporting.

**Comment by jpountz** [comment]
@jakelandis I bumped the version since this hasn't been backported yet.

**Comment by jakelandis** [comment]
I am trying to stabilize these tests in master before backporting. I will remove the backport versions and track this via the meta issue #42409

**Pull request merged**
